src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/generator/AbstractQueryGenerator.java

```diff
@@ -70,8 +76,52 @@ private String generateQueryBody(@NonNull Criteria criteria, @NonNull List<Pair<
     @NonNull
     protected Pair<String, List<Pair<String, Object>>> generateQueryBody(@NonNull DocumentQuery query) {
         final List<Pair<String, Object>> parameters = new ArrayList<>();
-        final String queryString = this.generateQueryBody(query.getCriteria(), parameters);
+        String queryString = this.generateQueryBody(query.getCriteria(), parameters);
+
+        if (StringUtils.hasText(queryString)) {
+            queryString = "WHERE" + " " + queryString;
+        }
 
         return Pair.with(queryString, parameters);
     }
+
+    private String getParameter(@NonNull Sort.Order order) {
+        String parameter = "r." + order.getProperty();
+
+        Assert.isTrue(!order.isIgnoreCase(), "Ignore case is not supported");
+
+        if (order.isDescending()) {
+            parameter += " " + Constants.SQL_KEYWORD_DESC;
```

Comment:
My suggestion would be to define some template to generate the real query, rather than cancat the string